gulp/tasks/jekyll-build-preview.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        gitUtils = require('../util/gitUtils'),
+        childProcess = require('child_process'),
+        previewConfigFile = '_tmp_preview_config.yml',
+        previewSiteDir = '_preview_site';
+
+    gulp.task('jekyll-build-preview', function() {
+        var branch = gitUtils.currentBranch(),
+            previewBaseUrl = '/' + branch + '/';
+        // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+        childProcess.execSync(
```

Comment:
It looks like [this function can throw an error](https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options). Could you add some error handling?